packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -1,204 +1,130 @@
 import * as React from 'react';
+import { useEventCallback, ownerDocument } from '@material-ui/core/utils';
 import { ColDef } from '../../models/colDef';
-import { ScrollParams, useLogger } from '../utils';
-import { COL_RESIZE_START, COL_RESIZE_STOP, SCROLLING } from '../../constants/eventsConstants';
-import { findCellElementsFromCol, findDataContainerFromCurrent } from '../../utils';
-import { useStateRef } from '../utils/useStateRef';
+import { useLogger } from '../utils';
+import { COL_RESIZE_START, COL_RESIZE_STOP } from '../../constants/eventsConstants';
+import { findCellElementsFromCol } from '../../utils';
 import { ApiRef } from '../../models';
 
 const MIN_COL_WIDTH = 50;
-const MOUSE_LEFT_TIMEOUT = 1000;
-
-// TODO improve experience for last column
-export const useColumnResize = (
-  columnsRef: React.RefObject<HTMLDivElement>,
-  apiRef: ApiRef,
-  headerHeight: number,
-) => {
-  const logger = useLogger('useColumnResize');
 
-  const isResizing = React.useRef<boolean>(false);
-  const isLastColumn = React.useRef<boolean>(false);
-  const mouseLeftTimeout = React.useRef<any>();
-  const stopResizeEventTimeout = React.useRef<any>();
+export const useColumnResize = (columnsRef: React.RefObject<HTMLDivElement>, apiRef: ApiRef) => {
+  const logger = useLogger('useColumnResize');
+  const colDefRef = React.useRef<ColDef>();
+  const colElementRef = React.useRef<HTMLDivElement>();
+  const colCellElementsRef = React.useRef<NodeListOf<Element>>();
+  const initialOffset = React.useRef<number>();
+  const stopResizeEventTimeout = React.useRef<number>();
+
+  const updateWidth = (newWidth: number) => {
+    logger.debug(`Updating width to ${newWidth} for col ${colDefRef.current!.field}`);
+
+    colDefRef.current!.width = newWidth;
+
+    colElementRef.current!.style.width = `${newWidth}px`;
+    colElementRef.current!.style.minWidth = `${newWidth}px`;
+    colElementRef.current!.style.maxWidth = `${newWidth}px`;
+
+    colCellElementsRef.current!.forEach((element) => {
+      const div = element as HTMLDivElement;
+      div.style.width = `${newWidth}px`;
+      div.style.minWidth = `${newWidth}px`;
+      div.style.maxWidth = `${newWidth}px`;
+    });
+  };
+
+  const handleResizeMouseMove = useEventCallback((nativeEvent) => {
+    // Cancel move in case some other element consumed a mouseup event and it was not fired.
+    if (nativeEvent.buttons === 0) {
+      // @ts-expect-error fixed in v5
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
```

Comment:
` // eslint-disable-next-line` is not great. Should we disable the rule? As there is no point of linting if whenever it breaks we disable the line => it will lead to bad practices for other maintainers as they will do the same.
easy one to fix!

Reply:
In this case, we can solve it by swapping the order of the methods, great catch